lib/rubocop/cop/rspec/repeated_subject_call.rb

```diff
@@ -64,16 +70,12 @@ def on_top_level_group(node)
 
         private
 
-        def detect_offense(example_node, subject_node)
-          walker = subject_node
+        def detect_offense(subject_node)
+          return if subject_node.chained?
 
-          while walker.parent? && walker.parent != example_node.body
-            walker = walker.parent
+          subject_node.each_ancestor(:block) do |block|
 
-            if walker.block_type? && walker.method?(:expect)
-              add_offense(walker)
-              return
-            end
+            add_offense(block) if block.method?(:expect)
```

Comment:
One last optional suggestion for cases with nested ‘except’ - do we need to mark both, or the inner is fine? We could probably ‘each_ancestor.find’?
Completely optional.
All is good, mege at will and thanks once again!

Reply:
How about this? Let me know if I'm wrong.